### PR TITLE
Products api

### PR DESCRIPTION
**Pull request by cantwait**
A couple of operations from Product model exposed as json API,

please review it, I noted some files were removed out of the source but not sure whether it was me or because of `bundle install/update` command.

### CODE REVIEW

**Comment by carlosero** [comment]
@cantwait please change base branch to master, dev has not been used in a while

File: app/helpers/api_products_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module ApiProductsHelper
+end
```
Comment (carlosero) [inline app/helpers/api_products_helper.rb]: Please delete this file.

File: app/controllers/api/v1/api_products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+			name = params[:name].downcase
+		end
+		products = Product.where("lower(name) LIKE ?", "#{name}%")
+       	paginate json: products
+    end
+
+    def show
+        render json: Product.find(params:id)
+    end
+end
```
Comment (carlosero) [inline app/controllers/api/v1/api_products_controller.rb]: Please use two-space identation

File: app/controllers/api/v1/api_products_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  	 headers['Access-Control-Allow-Headers'] = '*'
+	 headers['Access-Control-Expose-Headers'] = 'Link,Per-Page,Total'
+ 	 headers['Access-Control-Max-Age'] = "1728000"
+	end
+    def index
+		name = ''
+		if(params[:name])
+			name = params[:name].downcase
+		end
+		products = Product.where("lower(name) LIKE ?", "#{name}%")
```
Comment (carlosero) [inline app/controllers/api/v1/api_products_controller.rb]: You can use `"lower(name) LIKE lower(?)"` and safely pass `"#{params[:name]}%"` directly

File: app/controllers/api/v1/api_products_controller.rb (hunk 2 above)
Comment (cantwait) [inline app/controllers/api/v1/api_products_controller.rb]: done with all changes @carlosero thanks.

**Pull request merged**